android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -491,24 +491,52 @@ constructor(
     gatewayModeHeaderValue: String? = null,
     searchPath: String,
     patientRelatedResourceTypes: MutableList<ResourceType>,
+    currentPage: Int = 1,
   ) {
-    val resultBundle =
-      if (gatewayModeHeaderValue.isNullOrEmpty()) {
-        fhirResourceDataSource.getResource(searchPath)
-      } else
-        fhirResourceDataSource.getResourceWithGatewayModeHeader(
-          gatewayModeHeaderValue,
-          searchPath,
-        )
+    var currentPageUrl = "$searchPath&_page=$currentPage&_count=4"
```

Comment:
we can remove the hard coded 4 to pick from the constants. during dev test mode you can change the constant value instead

Reply:
Oh okay. Lemme correct that